Release v0.1.0

yamamoto0104 opened the pull request:
#### What type of PR is this?

Add one of the following kinds:

* documentation



#### What this PR does / why we need it:
Create release branch for v0.1.0 and include CHANGELOG

eric-murray commented on code/API_definitions/number-recycling.yaml:
```diff
@@ -73,7 +73,7 @@ externalDocs:
   url: https://github.com/camaraproject/NumberRecycling
 servers:
-  - url: '{apiRoot}/number-recycling/vwip'
+  - url: '{apiRoot}/number-recycling/0.1.0-alpha.1'
```
See rules for API versioning [here](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#53-api-versions-throughout-the-release-process)
```suggestion
  - url: '{apiRoot}/number-recycling/0.1alpha1'
```

yamamoto0104 replied:
I think `0.1alpha1` can be `v0.1alpha1`

eric-murray commented on CHANGELOG.md:
```diff
@@ -0,0 +1,25 @@
+
+**Please be aware that the project will have frequent updates to the main branch. There are no compatibility guarantees associated with code in any branch, including main, until it has been released. For example, changes may be reverted before a release is published. For the best results, use the latest published release.**
+
+# v0.1.0
```
```suggestion
# r1.1
```

eric-murray commented on CHANGELOG.md:
```diff
@@ -0,0 +1,25 @@
+
+**This is the first alpha version of the KYC APIs.** 
+
+- API [definition](https://github.com/camaraproject/NumberRecycling/tree/main/code/API_definitions)
```
Link should be to release tag (r1.1) rather than main branch
```suggestion
- [number-recycling v0.1.0-alpha.1](https://github.com/camaraproject/NumberRecycling/blob/r1.1/code/API_definitions/number-recycling.yaml)
```

eric-murray commented on CHANGELOG.md:
```diff
@@ -0,0 +1,25 @@
+
+## What's changed
+
+* New API `Number Recycling`, v0.1.0:
```
```suggestion
* New API `Number Recycling`, v0.1.0-alpha.1:
```

eric-murray commented:
@yamamoto0104
I would suggest to approve and merge #20 before merging this PR so that the alpha version includes the updated error codes

yamamoto0104 commented:
Hello @eric-murray 
I'm fine with your proposal. 
we are revieing #20 internally and feedback later.

yamamoto0104 commented:
Hello @eric-murray 
I think `0.1alpha1` can be `v0.1alpha1`

yamamoto0104 commented:
Hello @eric-murray 

Thank you for your comments. I updated the link and added @camaraproject/release-management_maintainers.

> * Update links in the YAML that reference `main` to reference the release tag `r1.1` instead
>   
>   * e.g. link to Number_Recycling_scenario_1.png should be: `src="https://raw.githubusercontent.com/camaraproject/NumberRecycling/r1.1/documentation/API_documentation/assets/Number_Recycling_scenario_1.png"`
>   * Include @camaraproject/release-management_maintainers in the list of reviewers for this PR

@Masa8106  Could you merge #18 ?
> * Merge [Adding API-Readiness-Checklist #18](https://github.com/camaraproject/NumberRecycling/pull/18) so that the API readiness checklist is in the release

Masa8106 commented:
@yamamoto0104  Yes. I have merged PR#18. Please check it.

hdamker commented on README.md:
```diff
@@ -7,19 +7,23 @@
 
 # NumberRecycling
-Repository to describe, develop, document, and test the NumberRecycling API within the 
+Repository to describe, develop, document, and test the NumberRecycling API within the KnowYourCustomer (KYC) API family.
```
Propose the adapt to the terms defined in updated ProjectStructureAndRoles: 

```suggestion
API Repository to describe, develop, document, and test the NumberRecycling API within the KnowYourCustomer (KYC) API Sub Project.
```

maxl2287 commented on code/API_definitions/number-recycling.yaml:
typos:

- line 282: `Permission denided.` -> change to `Permission denied.`
- line 384: `Too Many Requestes` -> change to `Too Many Requests`

yamamoto0104 replied:
Thanks. I fixed them.

hdamker commented on CHANGELOG.md:
```diff
@@ -0,0 +1,25 @@
+
+# r1.1
+
+**This is the first alpha version of the KYC APIs.** 
```
```suggestion
**This is the first alpha version of the Number Recycling API.** 
```

hdamker commented on CHANGELOG.md:
```diff
@@ -0,0 +1,25 @@
+
+**This is the first alpha version of the KYC APIs.** 
+
+- [number-recycling v0.1.0-alpha.1](https://github.com/camaraproject/NumberRecycling/blob/r1.1/code/API_definitions/number-recycling.yaml)
```
Here is no link expected, the links are in the section for the number-recycling v0.1.0-alpha.1 API.

See https://github.com/camaraproject/ReleaseManagement/blob/main/documentation/CHANGELOG_TEMPLATE.md
for the template.

yamamoto0104 replied:
Thanks @hdamker 
CHANGELOG.md was updated based on the template.

hdamker replied:
@yamamoto0104 the review issue https://github.com/camaraproject/ReleaseManagement/issues/133 is actually assigned to @FrimanJan. And don't worry, the deadline only requires to have the PR provided until Friday for review by ReleaseManagement. That you have achieved 👍

Masa8106 reviewed (dismissed):
LGTM

Masa8106 commented:
@maxl2287 , @eric-murray , @grgpapadopoulos 
Would you review this for approval? As discussed in the KYC sub group meeting, I am waiting for your approval to confirm your review results and will merge it tomorrow.

hdamker commented:
> @maxl2287 , @eric-murray , @grgpapadopoulos
> Would you review this for approval? As discussed in the KYC sub group meeting, I am waiting for your approval to confirm your review results and will merge it tomorrow.

@yamamoto0104 It's good to collect the approvals from your codeowner colleagues. But please wait with your merge for the approval from ReleaseManagement. The deadline on Friday requires only to provide the PR for RM approval, not to publish the pre-release. Thanks!

tanjadegroot commented:
Hi, some questions to progress the review:

1) the API is presented with an alpha version --> for M3, the expected version is a release candidate, so 0.1.0-rc.1

2) For the Spring25 meta-release, your API needs to be aligned with the Spring25 releases of Commonalities and ICM. 
Have you done the relevant updates ? in particular on the Device object and error responses ?
If aligned with Spring25, also update:
- in the OAS yaml: change x-camara-commonalities: 0.4.0 --> camara-commonalities: 0.5  (only 2 version numbers x.y)
- in the readiness checklist, please align to the latest template from here: https://github.com/camaraproject/ReleaseManagement/blob/main/documentation/API-Readiness-Checklist.md
Per this template, the last column is renamed to "Reference information" and, in items 2 and 3, the Commonalities and ICM release should be added (can be just text or a link)

3) per the above checklist, for M3 you need to provide the basic Test_definition file

tanjadegroot commented on code/API_definitions/number-recycling.yaml:
```diff
@@ -64,7 +64,7 @@ info:
 
 
-  version: wip
+  version: 0.1.0-alpha.1
```
The expected version should be at least 0.1.0-rc1 for M3

tanjadegroot commented on code/API_definitions/number-recycling.yaml:
```diff
@@ -64,7 +64,7 @@ info:
 
-  version: wip
+  version: 0.1.0-alpha.1
   x-camara-commonalities: 0.4.0
```
For Spring25, this should be x-camara-commonalities: 0.5

tanjadegroot commented on code/API_definitions/number-recycling.yaml:
```diff
@@ -73,7 +73,7 @@ externalDocs:
   url: https://github.com/camaraproject/NumberRecycling
 servers:
-  - url: '{apiRoot}/number-recycling/vwip'
+  - url: '{apiRoot}/number-recycling/v0.1alpha1'
```
For M3: should be v0.1rc1

tanjadegroot commented on code/API_definitions/number-recycling.yaml:
Please check alignment with 
- Commonalities r2.1 or 2.2 (shortly available)
- ICM r2.1 or r2.2 (shortly available)

especially on changes to error responses and Device / PhoneNumber
see the release CHANGELOG files and also here: https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/51413119/Version+0.5.0+supporting+documentation

eric-murray replied:
#30 required for compliance with ICM r2.2
#33 required for compliance with Commonalities r2.2

tanjadegroot commented on code/API_definitions/number-recycling.yaml:
line 158: suggest renaming the property "phoneNumChanged" to "NumberRecycled" as it is not the number that was changed, but the attached subscriber / owner.

tanjadegroot reviewed (changes requested):
to pass M3, please
- ensure alignment to Spring25 Commonalities & ICM
- ensure the PR delivers a release candidate version of the API (agreed by the team)
- provide the basic test file

yamamoto0104 commented:
@hdamker Thanks.

yamamoto0104 commented:
Hello @eric-murray 
When I try sync-folk to Release r1.1 branch, the following error occurred. It is probably due to a conflict between the three PRs (#30 , #33  and #38 ) merged earlier (especially the part that removed the 429 error). As a countermeasure, is it better to create a new PR reflecting the corrections in this PR, or to discard it?

![image](https://github.com/user-attachments/assets/da0fc07e-f6f5-42b2-9566-236bfd290d38)

yamamoto0104 commented:
@eric-murray 
429 issue was resolved. Other issues are version and x-camara-commonalities.
If we reverted to the original version (verision: wip,  x-camara-commonalities.:0.4.0), this conflict error was resolved.

![image](https://github.com/user-attachments/assets/d5e6e888-47c1-4899-aaa2-123e329b5304)

maxl2287 commented:
> Hello @eric-murray When I try sync-folk to Release r1.1 branch, the following error occurred. It is probably due to a conflict between the three PRs (#30 , #33 and #38 ) merged earlier (especially the part that removed the 429 error). As a countermeasure, is it better to create a new PR reflecting the corrections in this PR, or to discard it (discard 28 commits)?
> 
> ![image](https://private-user-images.githubusercontent.com/176380962/412906431-da0fc07e-f6f5-42b2-9566-236bfd290d38.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk1MjEwNDcsIm5iZiI6MTczOTUyMDc0NywicGF0aCI6Ii8xNzYzODA5NjIvNDEyOTA2NDMxLWRhMGZjMDdlLWY2ZjUtNDJiMi05NTY2LTIzNmJmZDI5MGQzOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIxNFQwODEyMjdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01Njc5YzU3ZTU5MzgxNzdmOGY3ZWQzNTg0ODc1NmUyNjZiZTZiMTU4MDVkNmIwZjU1M2UzYjhjOGYxZmFmMzNiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.MEvUZFcIZNt7_eWibLFzLynq4-PTy_7TMfkGaTzYVsk)

I would just to a merge of the upstream/main into your yamamoto_release_v0.1.0 - branch.

Then you just need to take care to compare your commits with the ones from the main, but let the versions of your branch in it.

eric-murray commented:
I fixed the conflict. The following is correct for the release:
```
  version: 0.1.0-rc.1
  x-camara-commonalities: 0.5
```

yamamoto0104 commented:
@eric-murray 
Thanks a lot!

yamamoto0104 commented:
@hdamker and @tanjadegroot 
Could you merge this PR into repositories? The following PRs for Commonalities r2.2 and ICM r2.2 have already been merged.
- #38 
- #33 
- #30

hdamker commented:
> @hdamker and @tanjadegroot Could you merge this PR into repositories? The following PRs for Commonalities r2.2 and ICM r2.2 have already been merged.
> 
> * [Remove non mandatory errors following new commonalities #38](https://github.com/camaraproject/NumberRecycling/pull/38)
> * [Update x-correlator pattern in number-recycling.yaml #33](https://github.com/camaraproject/NumberRecycling/pull/33)
> * [Update ICM authorisation and authentication wording in number-recycling.yaml #30](https://github.com/camaraproject/NumberRecycling/pull/30)

@yamamoto0104 The needed steps to merge this PR are as following:
* @tanjadegroot to approve that her requested changes are done
* At least one of your fellow codeowners need to approve as well
* Then merge the PR and directly afterwards create the pre-release in GitHub

tanjadegroot reviewed (commented):
Great progress - ok on these changes.

waiting for the test file to be merged before.

tanjadegroot commented on README.md:
```diff
@@ -7,19 +7,23 @@
 ## Scope
 
 * Service API “NumberRecycling” (see [APIBacklog.md](https://github.com/camaraproject/APIBacklog/blob/main/documentation/APIbacklog.md))
 * The API provides the customer with the ability to:  
```
```suggestion
* The API provides the API consumer with the ability to:  
```

tanjadegroot commented on README.md:
```diff
@@ -7,19 +7,23 @@
-The repository has no (pre)releases yet, work in progress is within the main branch.
+* Note: Please be aware that the project will have updates to the main branch. There are no compatibility guarantees associated with code in any branch, including main, until a new release is created. For example, changes may be reverted before a release is created. For best results, use the latest available release.
+
+* **The latest pre-release of CAMARA Number Recycling is [r1.1](https://github.com/camaraproject/NumberRecycling/tree/r1.1)**. The Release Tag is [r1.1](https://github.com/camaraproject/NumberRecycling/releases/tag/r1.1).
```
```suggestion
* **The latest pre-release of CAMARA Number Recycling is [r1.1](https://github.com/camaraproject/NumberRecycling/releases/tag/r1.1)**. The Release Tag is [r1.1](https://github.com/camaraproject/NumberRecycling/releases/tag/r1.1).
```

tanjadegroot commented:
for next time: merge would have needed release management approval first, but it looks all fine 
so LGTM from Release Management

tanjadegroot commented:
I have check and what has been merges is OK. /LGTM from Release mnagament
@hdamker please helps the team to create the release

hdamker commented:
@yamamoto0104 I've created the release already to ensure that the links in your README.md are not going to 404. 

For your reference and the next time the steps are:

1) go to "Releases" (right column on repository main page)
2) "create a new release" (or "draft a new release")
3) "Choose a tag": "create new tag r1.1 on publish
4) "Target": main (default)
5) "Title": "r1.1" (release name)
6) "Generate release notes": that's a convenient function to see all PRs which were merged between two releases, but delete "What's changed" section and replace it in next step
7) Copy section for the release out of the CHANGELOG.md (as markdown code), without the headline of the release itself, into the release description
8) Ensure that you correctly selected "pre-release" (or "latest" for public releases)
9) Publish release :-)

yamamoto0104 commented:
@hdamker Thank you for your instruction.
We submitted PR#40 for r1.2 yesterday, Would it be okay to create release/tag for r1.2? Or should we create them after it has been merged?

> @yamamoto0104 I've created the release already to ensure that the links in your README.md are not going to 404.
> 
> For your reference and the next time the steps are:
> 
> 1. go to "Releases" (right column on repository main page)
> 2. "create a new release" (or "draft a new release")
> 3. "Choose a tag": "create new tag r1.1 on publish
> 4. "Target": main (default)
> 5. "Title": "r1.1" (release name)
> 6. "Generate release notes": that's a convenient function to see all PRs which were merged between two releases, but delete "What's changed" section and replace it in next step
> 7. Copy section for the release out of the CHANGELOG.md (as markdown code), without the headline of the release itself, into the release description
> 8. Ensure that you correctly selected "pre-release" (or "latest" for public releases)
> 9. Publish release :-)

hdamker commented:
> We submitted PR #40 for r1.2 yesterday, Would it be okay to create release/tag for r1.2? Or should we create them after it has been merged?

The release tag has to be on the merge commit of the release PR, so you can't create it upfront. I've seen the release PR for r1.2 and create the release review issue for it: https://github.com/camaraproject/ReleaseManagement/issues/179. Please wait there for review comments and/or the approval by Release Management for the PR.

yamamoto0104 commented:
@hdamker , OK, we'll create the release tag after approval of Release Management team.
